### PR TITLE
201 refactor singlevaluesummary tab

### PR DESCRIPTION
**Pull request by dvdhoz**

### CODE REVIEW

**Review by joshuarwillis — COMMENTED** [review]
Looks good @dvdhoz. Just a few comments regarding the use of `assert` and a few minor comments on the GUI.

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 1]
```diff
@@ -1,40 +1,66 @@
...
-        setActionListeners();
+
+        initiateSingleValueComboBoxListener();
     }
 
-    private Statistics getFirstStat()
-    {
+    private void initiateSingleValueComboBoxListener() {
+        singleValueDataTransformView.setSingleValueDataViewListener(summaryType -> {
+            if(computeButton.getActionListeners() != null) {
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:48]: agree, removed the conditional statement

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 2]
```diff
@@ -44,15 +70,18 @@ private String getFirstStatString() {
         float[] vals = getFirstTextFieldValue();
         if (StatisticsUITypeMap.map.get(stat) == StatisticUIType.TEXTBOX) {
             if (stat == Statistics.PERCENTILES) {
+                assert vals != null;
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:73]: agree, changed to `if(vals != null)`

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 3]
```diff
@@ -44,15 +70,18 @@ private String getFirstStatString() {
...
                 r = String.format("%.2f%% %s", vals[0] * 100, stat);
             } else if (stat == Statistics.MAXAVERAGEDURATION || stat == Statistics.MAXACCUMDURATION) {
+                assert vals != null;
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:76]: agree, changed to `if(vals != null)`

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 4]
```diff
@@ -44,15 +70,18 @@ private String getFirstStatString() {
...
                 r = String.format("%d hour %s", (int)vals[0], stat);
             }
             else {
                 r = stat.toString();
             }
         } else {
             if (stat == Statistics.CUMULATIVE) {
+                assert vals != null;
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:84]: agree, changed to `if(vals != null)`

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 5]
```diff
@@ -67,8 +96,10 @@ private String getSecondStatString() {
         float[] vals = getSecondTextFieldValue();
         if (StatisticsUITypeMap.map.get(stat) == StatisticUIType.TEXTBOX) {
             if (stat == Statistics.PERCENTILES) {
+                assert vals != null;
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:99]: agree, changed to `if(vals != null)`

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 6]
```diff
@@ -67,8 +96,10 @@ private String getSecondStatString() {
...
                 r = String.format("%.2f%% %s", vals[0] * 100, stat);
             } else if (stat == Statistics.MAXAVERAGEDURATION || stat == Statistics.MAXACCUMDURATION) {
+                assert vals != null;
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:102]: agree, changed to `if(vals != null)`

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 7]
```diff
@@ -134,26 +161,15 @@ private void writeLn(String output) {
...
-                    type = t;
-                    break;
-                }
-            }
-            setupStatComboBoxes(type);
-
-        });
+        setupStatComboBoxes(selectedSummaryType);
 
         statComboBox1.addActionListener(e ->
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:166]: Concur, combined the logic

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:73]: Actually, I don't need the null check. Removed from method

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:76]: Actually, I don't need the null check. Removed from method

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:84]: Actually, I don't need the null check. Removed from method

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:99]: Actually, I don't need the null check. Removed from method

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:102]: Actually, I don't need the null check. Removed from method

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 8]
```diff
@@ -134,26 +161,15 @@ private void writeLn(String output) {
...
                 textField1.setEditable(StatisticsUITypeMap.map.get((Statistics) (statComboBox1.getSelectedItem())) != StatisticUIType.CHECKBOX));
 
         statComboBox2.addActionListener(e ->
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:169]: In the `setTextboxEditable()` method, set the text field to empty string

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 9]
```diff
@@ -256,39 +326,39 @@ private void setupStatComboBox2(SingleValueSummaryType option) {
     }
 
     private void initializeUI() {
-        leftPanel = new JPanel();
-        rightPanel = new JPanel();
+        topPanel = new JPanel();
+        bottomPanel = new JPanel();
+
+        label1 = new JLabel("1.");
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:332]: I like it.  Added method `setComputeOrderLabel` to SingleVaueDataViewListener interface to change the label so the user knows which compute is happening first

File: ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java [hunk 10]
```diff
@@ -176,64 +192,118 @@ private void setActionListeners() {
...
+        comboDim.height = 30;
+        statComboBox1.setPreferredSize(comboDim);
+        statComboBox2.setPreferredSize(comboDim);
+
+        gc.gridx = 0;
+        gc.gridy = 0;
+        gc.weightx = .1;
+
+        gc.anchor = GridBagConstraints.LINE_END;
+        selectionPanel.add(label1, gc);
```
[the remark this answers is not in the document]
Comment (dvdhoz) [reply ensemble-view/src/main/java/hec/ensembleview/tabs/SingleValueSummaryTab.java:258]: Added insets to the GridBagConstraints.  Also fixed the borders so they now align.

**Pull request merged**